src/SimplePie.php

```diff
@@ -817,6 +833,14 @@ public function set_timeout(int $timeout = 10)
      */
     public function set_curl_options(array $curl_options = [])
     {
+        if ($this->http_client_injected) {
+            throw new SimplePieException(sprintf(
+                'Using "%s()" has no effect, because you already provided a HTTP client with "%s::set_http_client()". Configure custom curl options in your HTTP client instead.',
+                __METHOD__,
+                self::class
+            ));
+        }
+
```

Comment:
We should unset a possible created $this->http_client on calling `set_curl_options()`.
```suggestion

        // Reset a possible existing FileClient,
        // so a new client with the changed value will be created
        $this->http_client = null;
```